internal/provider/types.go

```diff
@@ -57,3 +87,42 @@ func SupportedECDSACurvesStr() []string {
 	}
 	return supportedStr
 }
+
+// PEMPreamblePrivateKey represents the "type" heading used by a PEM-formatted Private Key.
+// See: https://datatracker.ietf.org/doc/html/rfc1421
+type PEMPreamblePrivateKey string
+
+const (
+	PrivateKeyRSA     PEMPreamblePrivateKey = "RSA PRIVATE KEY"
+	PrivateKeyECDSA   PEMPreamblePrivateKey = "EC PRIVATE KEY"
+	PrivateKeyED25519 PEMPreamblePrivateKey = "PRIVATE KEY"
+)
```

Comment:
Should `internal/provider/resource_private_key.go` also be using these new constants?